services/spar/src/Spar/App.hs

```diff
@@ -320,17 +340,21 @@ verdictHandlerResultCore bindCky = \case
       -- race conditions: if the user has been created on spar, but not on brig, 'getUser'
       -- returns 'Nothing'.  this is ok assuming 'createUser', 'bindUser' (called below) are
       -- idempotent.
-
-      case (viaBindCookie, viaSparCass) of
+      viaSparCassOldIssuer <-
+        if isJust viaSparCass
+          then pure Nothing
+          else findUserWithOldIssuer userref
+      case (viaBindCookie, viaSparCass, viaSparCassOldIssuer) of
         -- This is the first SSO authentication, so we auto-create a user. We know the user
         -- has not been created via SCIM because then we would've ended up in the
         -- "reauthentication" branch, so we pass 'ManagedByWire'.
-        (Nothing, Nothing) -> autoprovisionSamlUser userref Nothing ManagedByWire
+        (Nothing, Nothing, Nothing) -> autoprovisionSamlUser userref Nothing ManagedByWire
+        (Nothing, Nothing, Just (oldUserRef, uid)) -> moveUserToNewIssuer oldUserRef userref uid >> pure uid
         -- SSO reauthentication
-        (Nothing, Just uid) -> pure uid
+        (Nothing, Just uid, _) -> pure uid
         -- Bind existing user (non-SSO or SSO) to ssoid
-        (Just uid, Nothing) -> bindUser uid userref
-        (Just uid, Just uid')
+        (Just uid, Nothing, _) -> bindUser uid userref
```

Comment:
don't we need to take into account the moved issuer here?

Reply:
b9efae06695c19e13188ac24c96e9bf6e6efd1f6